consensus/consensus.go

```diff
@@ -211,6 +227,29 @@ func (consensus *Consensus) checkConsensusMessage(message consensus_proto.Messag
 	// check consensus Id
 	if consensusID != consensus.consensusID {
 		utils.GetLogInstance().Warn("Wrong consensus Id", "myConsensusId", consensus.consensusID, "theirConsensusId", consensusID, "consensus", consensus)
+		if consensus.IsLeader {
+			return false
+		}
+		// validator send consensus block to state syncing
+		if val, ok := consensus.blocksReceived[consensusID]; ok {
```

Comment:
I would suggest move this logic to the caller level of checkConsensusMessage, by defining and returning different error types and checking the type on the caller side.

Reply:
Sure